examples/emodel_pipeline_local_python/monitor_optimisation.py

```diff
@@ -18,12 +58,15 @@ def monitor_optimisation():
         print("No checkpoints found.")
 
     for path in paths:
-        data = pickle.load(open(path, "rb"))
-        seed = path.split("_seed=")[1].split(".pkl")[0]
-        generation = data["logbook"].select("gen")[-1]
-        best_score = sum(data["halloffame"][0].fitness.values)
-        print(f"Seed: {seed}, Generation: {generation}, Score: {best_score}")
-        best_fitness.append(sum(data["halloffame"][0].fitness.values))
+        with open(path, "rb") as file:
+            data = pickle.load(file)
+            seed = path.split("_seed=")[1].split(".pkl")[0]
```

Comment:
you can put just the pickle load in the scope of the with open. The rest of the code can be unindented